### PR TITLE
Distinguish CI command runs

### PR DESCRIPTION
**Pull request by fotanus**
This patch add CI information on the user_agent in order for us to track if a given command was run by a CI, as discussed in #3233.

### CODE REVIEW

**Comment by fotanus** [comment]
This needs to accepted along with [this PR on bundler-api](https://github.com/bundler/bundler-api/pull/75)

File: lib/bundler/fetcher.rb [hunk 1]
```diff
@@ -178,6 +180,23 @@ def inspect
...
+        "HAS_JOSH_K_SEAL_OF_APPROVAL" => "josh_k_seal",
+        "CIRCLECI" => "circle",
+        "SEMAPHORE" => "semaphore",
+        "JENKINS_URL" => "jenkins",
+        "BUILDBOX" => "buildbox",
+        "GO_SERVER_URL" => "go",
+        "SNAP_CI" => "snap"
+      }
+      cis = env_cis.find_all{ |env, ci| ENV[env]}.map{ |env, ci| ci }
+      cis << ENV["CI_NAME"] if ENV["CI_NAME"]
```
Comment (segiddins) [inline lib/bundler/fetcher.rb]: why aren't these in the hash?

```ruby
{
  "CI_NAME" => ENV["CI_NAME"],
  "CI" => "ci",
```

Comment (fotanus) [inline lib/bundler/fetcher.rb]: makes sense, will change it :+1:

**Comment by fotanus** [comment]
Oh, well, it seems that it is not passing on travis, because this is all about CIs. Let me rewrite this test to be more generic...

**Comment by indirect** [comment]
Happy to include this in the 1.10 release, as long as the specs are passing before we are ready to release. :)

**Comment by andremedeiros** [comment]
Actually, this is failing because you're adding `josh_k_seal`. Take that out and I'm sure it will pass ;-)

**Comment by andremedeiros** [comment]
To give you context, using that environment variable to detect Travis was more of a joke than anything, and we should probably not use it :-P

**Comment by fotanus** [comment]
@andremedeiros I wonder how I managed to break all that in my last commit? o.O

**Comment by andremedeiros** [comment]
Leave it to me. I'll have a look!

**Comment by fotanus** [comment]
closing this in favor of #3646

**Pull request closed without merging**